replication-api-impl/src/test/java/com/connexta/replication/api/impl/persistence/spring/FilterIndexRepositoryIntegrationTest.java

```diff
@@ -0,0 +1,281 @@
+import com.connexta.replication.api.impl.persistence.pojo.FilterIndexPojo;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import org.apache.solr.client.solrj.SolrClient;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.data.solr.UncategorizedSolrException;
+import org.springframework.data.solr.repository.config.EnableSolrRepositories;
+import org.springframework.data.solr.server.SolrClientFactory;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@ComponentScan(basePackageClasses = FilterIndexRepository.class)
+@RunWith(SpringRunner.class)
+public class FilterIndexRepositoryIntegrationTest {
+  private static final String ID1 = "id";
+  private static final Instant MODIFIED_SINCE1 = Instant.ofEpochSecond(100);
+  private static final String FILTER_ID1 = "filterId";
+  private static final FilterIndexPojo POJO1 =
+      new FilterIndexPojo().setId(ID1).setModifiedSince(MODIFIED_SINCE1).setFilterId(FILTER_ID1);
+
+  private static final String ID2 = "id2";
+  private static final Instant MODIFIED_SINCE2 = Instant.ofEpochSecond(200);
+  private static final String FILTER_ID2 = "filterId2";
+  private static final FilterIndexPojo POJO2 =
+      new FilterIndexPojo().setId(ID2).setModifiedSince(MODIFIED_SINCE2).setFilterId(FILTER_ID2);
+
+  private static final String ID3 = "id3";
+  private static final Instant MODIFIED_SINCE3 = Instant.ofEpochSecond(300);
+  private static final String FILTER_ID3 = "filterId3";
+  private static final FilterIndexPojo POJO3 =
+      new FilterIndexPojo().setId(ID3).setModifiedSince(MODIFIED_SINCE3).setFilterId(FILTER_ID3);
+
+  @TestConfiguration
+  @EnableSolrRepositories(basePackages = "com.connexta.replication")
+  static class ReplicationFilterIndexRepositoryTestConfiguration {
+    @Bean
+    SolrClientFactory solrFactory() {
+      return new EmbeddedSolrServerFactory("classpath:solr", FilterIndexPojo.COLLECTION);
+    }
+
+    @Bean
+    SolrClient solrClient(SolrClientFactory solrFactory) {
+      return solrFactory.getSolrClient();
+    }
+  }
+
+  @Rule public ExpectedException exception = ExpectedException.none();
+
+  @Autowired private FilterIndexRepository repo;
+
+  @After
+  public void cleanup() {
+    repo.deleteAll();
+  }
+
+  @Test
+  public void testPojoPersistenceWhenIdIsNotDefined() {
+    exception.expect(UncategorizedSolrException.class);
+
+    final FilterIndexPojo pojo =
+        new FilterIndexPojo().setModifiedSince(MODIFIED_SINCE1).setFilterId(FILTER_ID1);
+
+    repo.save(pojo);
+  }
+
+  @Test
```

Comment:
❗️ Add a test to verify that we can overwrite an existing index with a new one that changed its modified time

Reply:
Is that something we really need to verify? Can we not trust that solr has already tested the case of re-saving a config that already exists?

Reply:
This is just to make sure that we setup the schema properly and all.